sdk/tests/conformance2/query/occlusion-query.html

```diff
@@ -0,0 +1,157 @@
+** distribute, sublicense, and/or sell copies of the Materials, and to
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL Occlusion Query Conformance Tests</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<canvas id="canvas" style="width: 50px; height: 50px;"> </canvas>
+<div id="console"></div>
+<script>
+"use strict";
+description("This test verifies the functionality of occlusion query objects.");
+
+debug("");
+
+var tests = [];
+var currentTest;
+var currentTestIndex = 0;
+var numberOfTestAttempts = 4; // Just to stress implementations a bit more.
+var query;
+var numberOfCompletionAttempts = 0;
+
+function setupTests(gl) {
+    tests = [
+        {
+            target: gl.ANY_SAMPLES_PASSED_CONSERVATIVE,
+            name:   "ANY_SAMPLES_PASSED_CONSERVATIVE",
+            result: 1,
+        },
+        {
+            target: gl.ANY_SAMPLES_PASSED,
+            name:   "ANY_SAMPLES_PASSED",
+            result: 1,
+        },
+    ];        
+}
+
+function runOcclusionQueryTest() {
+    currentTest = tests[currentTestIndex];
+
+    debug("");
+    debug("Testing completion and behavior of " + currentTest.name + " occlusion query");
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+    var program = wtu.setupSimpleColorProgram(gl, 0);
+    gl.uniform4f(gl.getUniformLocation(program, "u_color"), 0, 1, 0, 1);
+    wtu.setupUnitQuad(gl, 0);
+    query = gl.createQuery();
+    var target = currentTest.target;
+    gl.beginQuery(target, query);
+    wtu.drawUnitQuad(gl);
+    gl.endQuery(target);
+
+    // Verify as best as possible that the implementation doesn't
+    // allow a query's result to become available the same frame, by
+    // spin-looping for some time and ensuring that none of the
+    // queries' results become available.
+    var numEarlyTests = 20000;
+    while (--numEarlyTests > 0) {
+        gl.finish();
+        if (gl.getQueryParameter(query, gl.QUERY_RESULT_AVAILABLE)) {
+            testFailed("Query's result became available too early");
+            return;
```

Comment:
Done in upcoming patch set. (Also in the same place in the transform_feedback.html test.)